CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+
+### The Gonum Repos
+
+Here are the current repos for the gonum project.
+If code you want to contribute doesn't quite fit in any of them, then please start a discussion on the [mailing list](https://groups.google.com/forum/#!forum/gonum-dev).
+
+* [matrix](https://github.com/gonum/matrix) - Matrix packages for the Go language.
+* [plot](https://github.com/gonum/plot) - A repository for plotting and visualizing data
+* [stat](https://github.com/gonum/stat) - Statistics package for Go
+* [blas](https://github.com/gonum/blas) - A blas implementation for Go.
+* [lapack](https://github.com/gonum/lapack) - A lapack implementation for go
+* [floats](https://github.com/gonum/floats) - Operations on slices of floats
+* [integrate](https://github.com/gonum/integrate) - integrate implements routines for numerical integration for the Go language
+* [diff](https://github.com/gonum/diff) - Set of tools for computing derivatives of a function
+* [graph](https://github.com/gonum/graph) - Graph packages for the Go language
+* [optimize](https://github.com/gonum/optimize) - Package for finding the optimum value of functions.
+* [unit](https://github.com/gonum/unit) - Package for converting between scientific units
+* [mathext](https://github.com/gonum/mathext) - mathext implements basic elementary functions not included in the Go standard library
+* [internal](https://github.com/gonum/internal) - Internal routines for the gonum project
+* [license](https://github.com/gonum/license) - Master license, author, contributor etc information.
+* [talks](https://github.com/gonum/talks) - Talks given about Go and Gonum delivered by Gonum developers.
+* [gonum.github.io](https://github.com/gonum/gonum.github.io) - Gonum website
+
+## Contributing
+
+### Reporting Bugs
+
+When you encounter a bug, please open an issue on the corresponding repo.
+Start the issue title with the repo/subrepo name, like `stat/distmv: Issue Name`.
+Be specific about the environment you encountered the bug in.
+If you are able to write a test that reproduces the bug, please include it in the issue.
+As a rule we keep all tests OK.
+
+### Suggesting Enhancements
+
+If the scope of the enhancement is small, open an issue.
+If it is large, such as suggesting a new repo, sub-repo, or interface refactoring, then please start a discussion on [the gonum-dev list](https://groups.google.com/forum/#!forum/gonum-dev).
+
+### Your First Code Contribution
+
+If you are a new contributor, thank you!  Before your first merge, you will need to be added to the [CONTRIBUTORS](https://github.com/gonum/license/blob/master/CONTRIBUTORS) and [AUTHORS](https://github.com/gonum/license/blob/master/AUTHORS) file.
+Open a pull request adding yourself to them.
+All Gonum code follows the BSD license in the license document.
+We prefer that code contributions do not come with additional licensing.
```

Comment:
I find these two sentences strange. Aren't they saying almost the same thing? Also, what are the criteria for allowing non-BSD license? And if we allow, then which licenses are ok? I would like to know myself.